client/components/image-preloader/index.tsx

```diff
@@ -0,0 +1,110 @@
+/**
+ * External dependencies
+ */
+import React, { useState, useEffect, useRef } from 'react';
+import { noop } from 'lodash';
+
+/**
+ * Constants
+ */
+const LoadStatus = {
```

Comment:
This looks like an [`enum`]():

```suggestion
enum LoadStatus {
```

I don't think we _need_ a string enum, so we can let TypeScript handle it:

```ts
enum LoadStatus {
	PENDING,
	LOADING,
	LOADED,
	FAILED,
};
```

Reply:
I keep forgetting about these, thank you!

The only potential issue I see with this is making things harder to debug in React DevTools, but I don't think it's much of an issue with a 4 item enum.